app/jobs/data_exports/create_job.rb

```diff
@@ -99,38 +122,75 @@ def create_attachment_manifest_file(attachment)
+    def create_analysis_zip(data_export)
+      new_zip_file = Tempfile.new(binmode: true)
+      ZipKit::Streamer.open(new_zip_file) do |zip|
+        workflow_execution = WorkflowExecution.find(data_export.export_parameters['ids'][0])
+        write_workflow_execution_outputs(workflow_execution, zip)
 
-        write_attachment(project.puid, sample.puid, zip, attachment)
+        sample_workflow_executions = workflow_execution.samples_workflow_executions
+        sample_workflow_executions.each do |swe|
+          write_sample_workflow_execution_outputs(swe, zip) if swe.outputs.count.positive?
+        end
+
+        update_analysis_manifest(workflow_execution, sample_workflow_executions)
+        # Write manifest to file 'manifest.json' and add to zip
+        write_manifest(zip)
+      end
+      new_zip_file
+    end
 
-        next unless attachment.metadata.key?('associated_attachment_id')
+    def update_analysis_manifest(workflow_execution, sample_workflow_executions)
+      add_workflow_execution_outputs_to_manifest(workflow_execution)
 
-        write_attachment(project.puid, sample.puid, zip, attachment.associated_attachment)
+      add_sample_workflow_executions_to_manifest(sample_workflow_executions)
+    end
+
+    def write_workflow_execution_outputs(workflow_execution, zip)
+      workflow_execution.outputs.each do |output|
+        write_attachment(output.file.filename.to_s, zip, output)
       end
     end
 
-    def write_attachment(project_puid, sample_puid, zip, attachment)
-      directory = "#{project_puid}/#{sample_puid}/#{attachment.puid}/#{attachment.file.filename}"
-      zip.write_file(directory) do |writer_for_file|
-        attachment.file.download { |chunk| writer_for_file << chunk }
+    def write_sample_workflow_execution_outputs(swe, zip)
+      sample = Sample.find(swe.sample_id)
+      swe.outputs.each do |output|
+        directory = "#{sample.puid}/#{output.file.filename}"
+        write_attachment(directory, zip, output)
       end
     end
 
-    def write_manifest(zip)
-      manifest_file = Tempfile.new
-      manifest_file.write(JSON.pretty_generate(JSON.parse(@manifest.to_json)))
-      zip.write_file('manifest.json') { |writer_for_file| IO.copy_stream(manifest_file.open, writer_for_file) }
+    def add_workflow_execution_outputs_to_manifest(workflow_execution)
+      workflow_execution.outputs.each do |workflow_execution_output|
+        @manifest['children'] << { 'name' => workflow_execution_output.file.filename.to_s, 'type' => 'file' }
+      end
+    end
 
-      manifest_file.close
-      manifest_file.unlink
+    def add_sample_workflow_executions_to_manifest(sample_workflow_executions)
+      sample_workflow_executions.each do |sample_workflow_execution|
+        next unless sample_workflow_execution.outputs.count.positive?
+
+        sample = Sample.find(sample_workflow_execution.sample_id)
```

Comment:
**nitpick (blocking):** If you update the query as commented above you can just do `sample_workflow_execution.sample`

Reply:
Added in 735c6182225cfb1787142006bed5fcf352e61fb3